presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2751,14 +2754,39 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
             predicate = createPredicate(partitionColumns, partitions);
         }
 
+        // Expose ordering property of the table.
+        ImmutableList.Builder<LocalProperty<ColumnHandle>> localProperties = ImmutableList.builder();
+        Optional<Set<ColumnHandle>> streamPartitionColumns = Optional.empty();
+        if (table.getStorage().getBucketProperty().isPresent() && isStreamingAggregationEnabled(session)) {
+            HiveBucketProperty bucketProperty = table.getStorage().getBucketProperty().get();
+            if (!bucketProperty.getSortedBy().isEmpty()) {
+                ImmutableSet.Builder<ColumnHandle> streamPartitionColumnsBuilder = ImmutableSet.builder();
+                Map<String, ColumnHandle> columnHandles = hiveColumnHandles(table).stream()
+                        .collect(toImmutableMap(HiveColumnHandle::getName, identity()));
+
+                partitionColumns.forEach(streamPartitionColumnsBuilder::add);
+                bucketProperty.getSortedBy().forEach(sortingColumn -> {
+                    ColumnHandle columnHandle = columnHandles.get(sortingColumn.getColumnName());
+                    localProperties.add(new SortingProperty<>(columnHandle, sortingColumn.getOrder().getSortOrder()));
+                    streamPartitionColumnsBuilder.add(columnHandle);
+                });
+
+                if (bucketProperty.getSortedBy().stream().map(SortingColumn::getColumnName).collect(toImmutableList()).equals(bucketProperty.getBucketedBy())) {
+                    // This should only be set when the all rows of the same value group are guaranteed to be in the same split. We only set this when the bucket columns are
```

Comment:
// This should only be set when all rows of the same value group are guaranteed to be in the same split. 
// We currently only set this for the cases where we want to enable streaming aggregation, i.e., when the bucket columns are the same as the sort columns. In this case, we also disable splitting a file for make sure the property is guaranteed.